mlflow/shap.py

```diff
@@ -240,8 +241,9 @@ def predict_function(X) -> pred:
     import shap
 
     artifact_path = _DEFAULT_ARTIFACT_PATH if artifact_path is None else artifact_path
-    background_data = shap.kmeans(features, min(_MAXIMUM_BACKGROUND_DATA_SIZE, len(features)))
-    explainer = shap.KernelExplainer(predict_function, background_data)
+    with mlflow.utils.autologging_utils.global_disable_autologging():
+        background_data = shap.kmeans(features, min(_MAXIMUM_BACKGROUND_DATA_SIZE, len(features)))
+        explainer = shap.KernelExplainer(predict_function, background_data)
     shap_values = explainer.shap_values(features)
```

Comment:
Can we put lines 247 - 256 inside the context manager as well? Perhaps `explainer.shap_values()` or `shap.summary_plot()` may make use of sklearn in the future.

Reply:
Yup sounds good